Monika After Story/game/cardgames.rpy

```diff
@@ -0,0 +1,5639 @@
+                                    self.chosen_colour = self.choose_colour(ignored_card=card)
+
+                                return card
+
+                # Come here when we don't want to/can't play a card
+                return None
+
+            def play_card(self, card):
+                """
+                Inner wrapper around play_card
+                NOTE: we do only certain checks here
+
+                IN:
+                    card - card to play
+                """
+                if not card:
+                    return
+
+                if card is self.queued_card:
+                    self.queued_card = None
+
+                self.game.play_card(self, self.game.player, card)
+
+                self.game.__win_check(self)
+
+                if (
+                    self.game.discardpile[-1].type == "wild"
+                ):
+                    self.game.discardpile[-1].colour = self.chosen_colour
+                    self.chosen_colour = None
+
+            def choose_reaction(self, next_card_to_play):
+                """
+                Helps Monika choose a dialogue based on the state of the game
+                NOTE: 'NOU' is handled differently, right in announce_reaction(), w/o corresponding reactions from here
+
+                TODO: reaction when you both are drawing cards
+                    because no one has a card with the current colour
+                TODO: reactions when Monika reflected a card on her 1st turn
+                    (the player had (had not) to skip their turn)
+                TODO: reactions when the player reflected a card on their 1st turn
+
+                IN:
+                    next_card_to_play - the next card Monika's going to play
+                        (we base reaction on it)
+                """
+                # # # Monika is going to reflect an action card
+                if (
+                    next_card_to_play is not None
+                    and self.should_skip_turn
+                    and len(self.game.game_log) > 1# see if there were enough turns
+                    and self.game.game_log[-2]["played_card"] is not None# check if the player played something
+                    and next_card_to_play.label == self.game.game_log[-2]["played_card"].label# compare the labels as this's the only way to reflect an act
+                ):
+                    # # # Does Monika reflect a card that reflected a wd4 before?
+                    # Monika played a wd4 > the player reflected > Monika reflected
+                    if (
+                        len(self.game.game_log) > 2
+                        and self.game.game_log[-3]["played_card"] is not None# see if Monika played wd4 before
+                        and self.game.game_log[-3]["played_card"].label == "Wild Draw Four"
+                    ):
+                        reaction = {
+                            "type": self.game.MONIKA_REFLECTED_WDF,
+                            "seen_count": 0
```

Comment:
For consistency with `Event` and ergo familiarity for testing purposes/debug/re-reading, we should make this `shown_count`